File: salesforge/backend/app/schemas/activity.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ActivityCreate(BaseModel):
    type: str
    title: str
    description: str | None = None
    due_date: datetime | None = None
    deal_id: str | None = None
    contact_id: str | None = None
    lead_id: str | None = None


class ActivityUpdate(BaseModel):
    type: str | None = None
    title: str | None = None
    description: str | None = None
    due_date: datetime | None = None
    completed: bool | None = None
    deal_id: str | None = None
    contact_id: str | None = None
    lead_id: str | None = None


class ActivityResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    type: str
    title: str
    description: str | None
    due_date: datetime | None
    completed: bool
    deal_id: str | None
    contact_id: str | None
    lead_id: str | None
    owner_id: str
    created_at: datetime
    updated_at: datetime


class ActivityListResponse(BaseModel):
    items: list[ActivityResponse]
    total: int
    page: int
    page_size: int

File: salesforge/frontend/src/types/activity.ts
export type ActivityType = 'call' | 'email' | 'meeting' | 'note' | 'task'

export interface Activity {
  id: string
  type: string
  title: string
  description: string | null
  due_date: string | null
  completed: boolean
  deal_id: string | null
  contact_id: string | null
  lead_id: string | null
  owner_id: string
  created_at: string
  updated_at: string
}

export interface ActivityListResponse {
  items: Activity[]
  total: number
  page: number
  page_size: number
}

export interface ActivityCreate {
  type: string
  title: string
  description?: string
  due_date?: string
  deal_id?: string
  contact_id?: string
  lead_id?: string
}

export type ActivityUpdate = Partial<ActivityCreate & { completed: boolean }>
